client/src/components/BidsModal.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import * as Dialog from '@radix-ui/react-dialog';
+import { Table, TableHeader, TableRow, TableHead, TableBody, TableCell } from '@/components/ui/table';
+import { Button } from '@/components/ui/button'
+import {useQuery } from '@tanstack/react-query'
+import { useMutation } from '@tanstack/react-query'
+import type { BidDtoReadable } from '@/api/generated';
+import { acceptAuctionMutation, acceptBidMutation, listBidsOptions } from '@/api/generated/@tanstack/react-query.gen'
+import { formatDate } from '@/lib/utils';
+
+interface BidsModalProps {
+  auctionId: number;
+  isOpen: boolean;
+  setIsOpen: (open: boolean) => void;
+  updateAuction: () => void;
+}
+
+const BidsModal: React.FC<BidsModalProps> = ({ auctionId, isOpen, setIsOpen, updateAuction }) => {
+  const { data, isLoading, isError } = useQuery(
+    {
+      ...listBidsOptions({ path: { auctionId: auctionId } }),
+      enabled: !!auctionId,
+    }
+  );
+
+  const bidsData = data as BidDtoReadable[];
```

Comment:
considère `as` comme le démon, à utiliser vraiment en cas de bug.

Ici le problème c'est que l'api swagger est mal annotée, du coup tu récupères unknow.

```suggestion
  const bidsData = data 
```